pkg/model/manifest.go

```diff
@@ -1206,3 +1206,68 @@ func getBuildContextForComposeWithVolumeMounts(m *Manifest) (string, error) {
 
 	return context, nil
 }
+
+func (di DeployInfo) IsEmpty() bool {
+	if len(di.Commands) > 0 {
+		return false
+	}
+	if di.ComposeSection != nil && di.ComposeSection.Stack != nil {
+		return false
+	}
+	if di.Divert != nil {
+		return false
+	}
+	if di.Endpoints != nil {
+		return false
+	}
+	return true
+}
+
+func (di DestroyInfo) IsEmpty() bool {
+	if di.Image != "" {
+		return false
+	}
+	if len(di.Commands) > 0 {
+		return false
+	}
+	return true
+}
+
+func (mt ManifestTests) IsEmpty() bool {
+	return len(mt) == 0
+}
+
+func (m *Manifest) ValidateForCLIOnly() error {
+	invalidFields := []string{}
```

Comment:
I am concerned this function will be missed to be updated when we introduced new top-level properties.

According to the ticket `dev` is the only property supported, if that is true, couldn't we check of any additional property different from `dev` instead of explicitly checking them?

Reply:
The problem with that is that we initialise some of them because otherwise we have nil pointers exceptions. Changing it to just check dev could be addressed but that will require a lot of changes and possible nil pointers that we can forget updating